sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParser.java

```diff
@@ -86,14 +88,31 @@ private void handleFileRef(XmlParserHelper xmlParserHelper) {
 
     private void handleFileTag(XmlParserHelper xmlParserHelper) {
       String uid = xmlParserHelper.getRequiredAttribute("uid");
-      String fullPath = xmlParserHelper.getRequiredAttribute("fullPath");
+      String pathInCoverageFile = xmlParserHelper.getRequiredAttribute("fullPath");
 
+      String canonicalPath;
       try {
-        files.put(uid, new File(fullPath).getCanonicalPath());
+        canonicalPath = new File(pathInCoverageFile).getCanonicalPath();
       } catch (IOException e) {
-        LOG.debug("Skipping the import of OpenCover code coverage for the invalid file path: " + fullPath
+        LOG.debug("Skipping the import of OpenCover code coverage for the invalid file path: " + pathInCoverageFile
           + " at line " + xmlParserHelper.stream().getLocation().getLineNumber(), e);
+        return;
       }
+
+      CoveredFile coveredFile;
+      if (fileService.isSupportedAbsolute(canonicalPath)) {
+        coveredFile = new CoveredFile(uid, pathInCoverageFile, canonicalPath);
+      } else {
+        // maybe it's a deterministic build file path
+        Optional<String> optionalPath = fileService.getAbsolutePath(pathInCoverageFile);
+        if (optionalPath.isPresent()) {
+          coveredFile = new CoveredFile(uid, pathInCoverageFile, optionalPath.get());
+          LOG.debug("Found indexed file '{}' for coverage entry '{}'.", coveredFile.supportedPath, coveredFile.originalPath);
+        } else {
+          coveredFile = new CoveredFile(uid, pathInCoverageFile, null);
+        }
+      }
+      files.put(uid, coveredFile);
```

Comment:
I was logging at debug level when we were resolving the deterministic path because that's only what I considered as needed to be logged.

If the path is not found at all, it's being logged in different places (for SequencePoint / BranchPoint), and now we have some redundancy - see the new test cases.

However, I'll keep this redundancy and see if it will bother us in practice.